backend/gn_module_monitoring/command/utils.py

```diff
@@ -116,6 +116,137 @@ def process_available_permissions(module_code, session):
         )
 
 
+def clean_existing_permissions(module_code, session):
+    try:
+        module = session.scalars(select(TModules).where(TModules.module_code == module_code)).one()
+    except NoResultFound:
+        print(f"Le module {module_code} n'est pas présent")
+        return
+    print(f" Reprise des permissions du module {module_code}")
+    # ########
+    # Mise à jour des permissions existantes vers les sous objets
+    # Création des permissions des sous-objets à partir des permissions contenus dans l'objet ALL
+    print(
+        " Création des permissions des sous-objets à partir des permissions contenus dans l'objet ALL"
+    )
+    session.execute(
+        text(
+            """
+        WITH ap AS (
+            SELECT o.code_object,o.id_object, tpa.id_module
+            FROM gn_permissions.t_permissions_available AS tpa
+            JOIN gn_permissions.t_objects AS o
+            ON o.id_object = tpa.id_object AND NOT code_object = 'ALL'
+            JOIN gn_commons.t_modules AS tm
+            ON tm.id_module = tpa.id_module AND tm.module_code = :module_code
+            JOIN gn_permissions.bib_actions AS ba
+            ON tpa.id_action = ba.id_action
+            WHERE NOT (code_object = 'MONITORINGS_MODULES' AND ba.code_action = 'U')
+        ), ep AS (
+                SELECT id_role, id_action, tp.id_module , tp.id_object, scope_value, sensitivity_filter
+                FROM gn_permissions.t_permissions AS tp
+                JOIN gn_permissions.t_objects AS o
+                ON o.id_object = tp.id_object  
+                JOIN gn_commons.t_modules AS tm
+                ON tm.id_module = tp.id_module AND tm.module_code = :module_code
+        ), new_p AS (
+            SELECT DISTINCT ep.id_role, ep.id_action, ep.id_module, ap.id_object, ep.scope_value, ep.sensitivity_filter
+            FROM ep
+            JOIN ap
+            ON ep.id_module = ap.id_module
+            LEFT OUTER JOIN  gn_permissions.t_permissions AS p
+            ON p.id_role = ep.id_role
+            AND  p.id_action = ep.id_action
+            AND  p.id_module = ep.id_module
+            AND  p.id_object = ap.id_object
+            WHERE p.id_permission IS NULL
+        )
+        INSERT INTO gn_permissions.t_permissions
+        (id_role, id_action, id_module, id_object, scope_value, sensitivity_filter)
+        SELECT id_role, id_action, id_module, id_object, scope_value, sensitivity_filter
+        FROM new_p;
+    """
+        ).bindparams(module_code=module_code)
+    )
```

Comment:
**Suggestion:** Consider using parameterized queries or ORM methods instead of string formatting to prevent SQL injection attacks. [security]
```suggestion
# Assuming SQLAlchemy ORM is used
# This is a simplified version, the actual code might need to be adjusted
with session.begin():
    ap = session.query(
        TObjects.code_object,
        TObjects.id_object,
        TPermissionsAvailable.id_module
    ).join(
        TPermissionsAvailable, TObjects.id_object == TPermissionsAvailable.id_object
    ).filter(
        TObjects.code_object != 'ALL',
        TModules.id_module == TPermissionsAvailable.id_module,
        TModules.module_code == module_code,
        ~((TObjects.code_object == 'MONITORINGS_MODULES') & (BibActions.code_action == 'U'))
    ).subquery()

    # Similar for other subqueries (ep, new_p)
    # ...
    
    # Then use these subqueries in the final query
    # ...
```